Sources/SwiftSyntaxBuilderGeneration/Templates/BuildableCollectionNodesFile.swift

```diff
@@ -0,0 +1,296 @@
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import SwiftSyntax
+import SwiftSyntaxBuilder
+
+let buildableCollectionNodesFile = SourceFile {
+  ImportDecl(
+    leadingTrivia: .docLineComment(copyrightHeader),
+    path: "SwiftSyntax"
+  )
+
+  for node in SYNTAX_NODES where node.isSyntaxCollection {
+    let type = node.type
+    let elementType = node.collectionElementType
+    let conformances = ["ExpressibleByArrayLiteral", "SyntaxBuildable", type.expressibleAs]
+    
+    // Generate collection node struct
+    StructDecl(
+      leadingTrivia: node.documentation.isEmpty
+        ? []
+        : .docLineComment("/// \(node.documentation)") + .newline,
+      modifiers: [TokenSyntax.public],
+      identifier: type.buildable,
+      inheritanceClause: createTypeInheritanceClause(conformances: conformances)
+    ) {
+      VariableDecl(.let, name: "elements", type: ArrayType(elementType: elementType.buildable))
+
+      // Generate initializers
+      createArrayInitializer(node: node)
+      createCombiningInitializer(node: node)
+      createArrayLiteralInitializer(node: node)
+
+      // Generate function declarations
+      createBuildFunction(node: node)
+      createBuildSyntaxFunction(node: node)
+      createExpressibleAsCreateFunction(node: node)
+      createSyntaxBuildableCreateFunction(node: node)
+    }
+
+    // For nodes without expressible-as conformances, conform Array to the corresponding expressible-as
+    if type.generatedExpressibleAsConformances.isEmpty {
+      ExtensionDecl(
+        extendedType: "Array",
+        inheritanceClause: TypeInheritanceClause {
+          InheritedType(typeName: type.expressibleAs)
+        },
+        genericWhereClause: GenericWhereClause(leadingTrivia: .space) {
+          GenericRequirement(body: SameTypeRequirement(
+            leftTypeIdentifier: "Element",
+            equalityToken: .spacedBinaryOperator("=="),
+            rightTypeIdentifier: elementType.expressibleAs
+          ))
+        }
+      ) {
+        FunctionDecl(
+          modifiers: [TokenSyntax.public],
+          identifier: .identifier("create\(type.buildableBaseName)"),
+          signature: FunctionSignature(
+            input: ParameterClause(),
+            output: type.buildable
+          )
+        ) {
+          ReturnStmt(expression: FunctionCallExpr(type.buildable) {
+            TupleExprElement(expression: "self")
+          })
+        }
+      }
+    }
+  }
+}
+
+/// Generate an initializer taking an array of elements.
+private func createArrayInitializer(node: Node) -> InitializerDecl {
+  let type = node.type
+  let elementType = node.collectionElementType
+  return InitializerDecl(
+    leadingTrivia: [
+      "/// Creates a `\(type.buildable)` with the provided list of elements.",
+      "/// - Parameters:",
+      "/// - elements: A list of `\(elementType.expressibleAs)`",
```

Comment:
The elements parameter needs to be indented so source editors will pick it up as a parameter documentation IIRC.

```suggestion
      "///   - elements: A list of `\(elementType.expressibleAs)`",
```